README.md

```diff
@@ -1,11 +1,39 @@
 # patch-filters
 special project to host GSOC project Patch Suppressions 
 
+## PatchFilter Description
+
+### SuppressionPatchFilter
+
+Filter SuppressionPatchFilter(Checker level) only accepts audit events for Check violations whose line number
+belong to added/changed lines in patch file and will suppress all Checks’ violation messages 
+which are not from added/changed lines. If there is no configured patch file 
+or the optional is set to true and patch file was not found the Filter suppresses all audit events.
+
+#### Properties
+ 
+ | name                  | description                                                  | type                                                         | default value |
+ | --------------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------- |
+ | file                  | Specify the location of the patch file.                      | [String](https://checkstyle.sourceforge.io/property_types.html#String) | Null          |
+ | optional              | Control what to do when the file is not existing. If `optional` is set to `false` the file must exist, or else it ends with error. On the other hand if optional is `true` and file is not found, the filter suppresses all audit events. | [boolean](https://checkstyle.sourceforge.io/property_types.html#boolean) | false         |
+ | strategy              | Control suppression scope that you need. If `startegy` is set to `newline`, it only accepts audit events for Check violations whose line number belong to added lines in patch file. `patchedline` will accept added/changed lines. | [String](https://checkstyle.sourceforge.io/property_types.html#String) | newline       |
+ | neverSuppressedChecks | String has user defined Checks to never suppress if files are touched, split by comma. This property is useful for Checks that place violation on whole file or on not all (first/last) occurrence of cause/violated code. | [String](https://checkstyle.sourceforge.io/property_types.html#String) | null          |
+
+#### Examples
+
+For example, the following configuration fragment directs the Checker to use a SuppressionPatchFilter with patch file config/defaultContext.patch:
+```xml
+<module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchFilter">
+    <property name="file" value="config/defaultContext.patch" />
+    <property name="strategy" value="newline" />
+</module>
+```
```

Comment:
Please add example for `neverSuppressedChecks`

Reply:
done